src/store/zoom.ts

```diff
@@ -0,0 +1,151 @@
+import { createSlice, PayloadAction, createSelector } from '@reduxjs/toolkit';
+import { RootState } from '../store';
+
+interface ZoomState {
+	byId: {
+		[zoomId: string]: {
+			zoomId: string;
+			discordId?: string;
+			name: string;
+		};
+	};
+	online: string[];
+	temporary: string[];
+	hasSeenStart: boolean;
+	active: boolean;
+	joinUrl: string;
+	lastZoomTime: number;
+}
+
+const initialState: ZoomState = {
+	byId: {},
+	online: [],
+	temporary: [],
+	hasSeenStart: false,
+	active: false,
+	joinUrl: '',
+	lastZoomTime: -1,
+};
+
+type UserEventAction = PayloadAction<{ zoomId: string; name: string }>;
+type UserLinkAction = PayloadAction<{ zoomId: string; discordId: string }>;
+
+const { reducer, actions, name } = createSlice({
+	name: 'zoomInfo',
+	reducers: {
```

Comment:
nit: it bothers me that some of the actions are past-tense (`thingHappened`) while some are imperative (`doThing`). I feel they should at least be consistent, and preferably imperative since they are actions. However, I don't have a good idea for how to name them as such. If you have an idea, then I think you should do that, but otherwise, _c'est la vie_.